docs/Modern-TypeEmission.md

```diff
@@ -0,0 +1,325 @@
+  artistID: 'banksy',
+}
+
+<QueryRenderer
+  query={graphql`
+    query ExampleQuery($artistID: ID!) {
+      artist(id: $artistID) {
+        name
+      }
+    }
+  `}
+  variables={variables}
+/>
+```
+<!--TypeScript-->
+```javascript
+/**
+ * export type ExampleQueryVariables = {
+ *   readonly artistID: string
+ * }
+ * export type ExampleQuery = {
+ *   readonly variables: ExampleQueryVariables
+ * }
+ */
+import { ExampleQuery } from "__generated__/ExampleQuery.graphql"
+
+<QueryRenderer<ExampleQuery>
+  query={graphql`
+    query ExampleQuery($artistID: ID!) {
+      artist(id: $artistID) {
+        name
+      }
+    }
+  `}
+  variables={{
+    artistID: 'banksy',
+  }}
+/>
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+### Operation/Fragment selection-set data
+
+The shape of the data selected in a operation or fragment, following the [data-masking] rules. That is, excluding any data selected by fragment spreads, unless the `@relay(mask: false)` directive is used.
+
+In this example the emitted type-information describes the response data available to the operation’s render function.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Flow-->
+```javascript
+/**
+ * export type ExampleQueryResponse = {|
+ *   +artist: ?{|
+ *     +name: string
+ *   |}
+ * |}
+ */
+import type { ExampleQueryResponse } from "__generated__/ExampleQuery.graphql"
+
+<QueryRenderer
+  query={graphql`
+    query ExampleQuery {
+      artist(id: "banksy") {
+        name
+      }
+    }
+  `}
+  render={({ props }: { props?: ExampleQueryResponse }) => {
+    if (props) {
+      return props.artist && <div>{props.artist.name} is great!</div>
+    }
+    return <div>Loading</div>
+  }}
+/>
+```
+<!--TypeScript-->
+```javascript
+/**
+ * export type ExampleQueryResponse = {
+ *   readonly artist?: {
+ *     readonly name: string
+ *   }
+ * }
+ * export type ExampleQuery = {
+ *   readonly response: ExampleQueryResponse
+ * }
+ */
+import { ExampleQuery } from "__generated__/ExampleQuery.graphql"
+
+<QueryRenderer<ExampleQuery>
+  query={graphql`
+    query ExampleQuery {
+      artist(id: "banksy") {
+        name
+      }
+    }
+  `}
+  render={({ props }) => {
+    if (props) {
+      return props.artist && <div>{props.artist.name} is great!</div>
+    }
+    return <div>Loading</div>
+  }}
+/>
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+Similarly, in this example the emitted type-information describes the prop data that the container expects to receive.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Flow-->
+```javascript
+/**
+ * export type ExampleFragment_artist = {|
+ *   +name: string
+ * |}
+ */
+import type { ExampleFragment_artist } from "__generated__/ExampleFragment_artist.graphql"
+
+export const ExampleFragment = createFragmentContainer(
+  (props: { artist: ExampleFragment_artist }) => (
+    <div>About the artist: {props.artist.biography}</div>
+  ),
+  graphql`
+    fragment ExampleFragment_artist on Artist {
+      biography
+    }
+  `
```

Comment:
let's change this to 

```
  {
    artist: graphql`...`,
  }
```

The fact that this syntax works is an unfortunate accident and we're going to move toward dropping support (first marking as deprecated, of course). We've received a lot of feedback that confirms our intuition that having multiple ways of defining fragments is confusing to users, and it also makes it harder to do things like have Flow understand the content of the tag (something we're working on :-) )

Reply:
oh nice, I've never seen this format - definitely reads better.

Reply:
Yeah, I always have that indecisive tension, so good to hear 👍